retriever/lib/engine_tools.py

```diff
@@ -101,37 +103,93 @@ def reset_retriever(scope="all", ask_permission=True):
     if output_file is None:
         output_file = os.path.splitext(os.path.basename(input_file))[0] + ".csv"
     csv_out = open_fw(output_file, encoding=encoding)
     if os.name == 'nt':
-        outfile = csv.DictWriter(csv_out,
-                                 dialect='excel',
-                                 escapechar="\\",
-                                 lineterminator='\n',
-                                 fieldnames=header_values)
+        outfile = csv.writer(csv_out,
+                             dialect='excel',
+                             escapechar="\\",
+                             lineterminator='\n')
     else:
-        outfile = csv.DictWriter(csv_out,
-                                 dialect='excel',
-                                 escapechar="\\",
-                                 fieldnames=header_values)
+        outfile = csv.writer(csv_out, dialect='excel', escapechar="\\")
     raw_data = json.loads(file_out.read())
-    outfile.writeheader()
-
-    for item in raw_data:
-        outfile.writerow(item)
+    raw_data, header_values = walker(raw_data,
+                      row_key=row_key,
+                      header_values=header_values,
+                      rows=[],
+                      normalize=False)
+
+    if isinstance(raw_data[0], dict):
+        raw_data = [list(row.values()) for row in raw_data]
+    else:
+        raw_data = [row.tolist() for row in raw_data]
+    if header_values:
+        outfile.writerow(header_values)
+    outfile.writerows(raw_data)
     file_out.close()
     subprocess.call(['rm', '-r', input_file])
     return output_file
 
 
+def walker(dictionary, row_key=None, header_values=None, rows=[], normalize=False):
+    """
+    Extract rows of data from json datasets
+    """
+    #  Handles the simple case, where row_key and column_key are not required
+    if not (row_key or header_values):
+        if isinstance(dictionary, dict):
+            rows = pd.DataFrame([dictionary]).values
+            header_values = dictionary.keys()
+            return rows, header_values
+        elif isinstance(dictionary, list):
+            rows = pd.DataFrame(dictionary, columns=header_values).values
+            header_values = [chr(i) for i in range(ord('a'),ord('a')+len(rows[0]))]
+            return rows, header_values
+
+    if isinstance(dictionary, dict):
+        if header_values and (set(header_values).issubset(dictionary.keys())):
+            if normalize:
+                rows.extend(
+                    json_normalize(
+                        dict(i for i in dictionary.items()
+                             if i[0] in header_values)).values)
+            else:
+                rows.extend(
+                    [dict(i for i in dictionary.items() if i[0] in header_values)])
+
+        elif dictionary.get(row_key):
+            if normalize:
+                rows.extend(json_normalize(dictionary[row_key]).values)
+            else:
```

Comment:
@ashishpriyadarshiCIC we forgot something, This should return two values not one 
`rows, header_values = walker(`
same applies to all the usages below